File: MyStagePass.Model/Models/Location.cs
using System.Collections.Generic;

namespace MyStagePass.Model.Models
{
    public class Location
    {
        public int LocationID { get; set; }
        public string LocationName { get; set; }
        public int Capacity { get; set; }
        public string Address { get; set; }
        public int CityID { get; set; }
        public virtual City City { get; set; } = null!;
        public virtual ICollection<Event> Events { get; set; } = new List<Event>();

    }
}

File: UI/mystagepass_admin/lib/models/Location/location.dart
import 'package:json_annotation/json_annotation.dart';
import '../City/city.dart';

part 'location.g.dart';

@JsonSerializable()
class Location {
  @JsonKey(name: 'locationID')
  int? locationId;
  String? locationName;
  int? capacity;
  String? address;
  @JsonKey(name: 'cityID')
  int? cityId;

  City? city;

  Location({
    this.locationId,
    this.locationName,
    this.capacity,
    this.address,
    this.cityId,
    this.city,
  });

  factory Location.fromJson(Map<String, dynamic> json) =>
      _$LocationFromJson(json);
  Map<String, dynamic> toJson() => _$LocationToJson(this);
}
